pylearn2/datasets/dense_design_matrix.py

```diff
@@ -1205,17 +1214,27 @@ def init_hdf5(self, path, shapes):
             The name of the hdf5 file.
         shapes : tuple
             The shapes of X and y.
+        title : string, optional
+            Name of the dataset. e.g. For SVHN, set this to "SVHN Dataset".
+            "Pytables Dataset" is used as title, by default.
+        y_dtype : string, optional
+            Either 'float' or 'int'. Decides the type of pytables atom
+            used to store the y data. By default 'float' type is used.
         """
 
         x_shape, y_shape = shapes
         # make pytables
         ensure_tables()
-        h5file = tables.openFile(path, mode="w", title="SVHN Dataset")
+        h5file = tables.openFile(path, mode="w", title=title)
         gcolumns = h5file.createGroup(h5file.root, "Data", "Data")
         atom = (tables.Float32Atom() if config.floatX == 'float32'
                 else tables.Float64Atom())
         h5file.createCArray(gcolumns, 'X', atom=atom, shape=x_shape,
                             title="Data values", filters=self.filters)
+        if y_dtype != 'float':
```

Comment:
Can you make sure that `y_dtype` is either `'float'` or `'int'` here? It may be confusing if a user passes `y_dtype = 'int32'` for instance,